packages/vm/benches/main.rs

```diff
@@ -135,8 +136,33 @@ fn bench_cache(c: &mut Criterion) {
             let _ = cache
                 .get_instance(&checksum, backend, DEFAULT_INSTANCE_OPTIONS)
                 .unwrap();
-            assert_eq!(cache.stats().hits_fs_cache, 1);
+            assert_eq!(cache.stats().hits_pinned_memory_cache, 0);
             assert!(cache.stats().hits_memory_cache >= 1);
+            assert_eq!(cache.stats().hits_fs_cache, 1);
+            assert_eq!(cache.stats().misses, 0);
+        });
+    });
+
+    group.bench_function("instantiate from pinned memory", |b| {
+        let checksum = Checksum::generate(CONTRACT);
+        let mut cache: Cache<MockApi, MockStorage, MockQuerier> =
+            unsafe { Cache::new(options.clone()).unwrap() };
+        // Load into memory
+        cache
+            .get_instance(&checksum, mock_backend(&[]), DEFAULT_INSTANCE_OPTIONS)
+            .unwrap();
```

Comment:
Isn't this done as part of `pin`? Then we we can simplify this benchmarking code